### PR TITLE
[runtime] Always release blocks on the main thread. Fixes #4130. (#4312)

### LINKED ISSUE #4130: Reproducible native crash bug with WKWebView in latest Xamarin.iOS
(Coming from Bugzilla we were at a loss where to post this and wrongly submitted this bug report to the Xamarin forums first, sorry about that).

Loading a WKWebView with a HTML string twice using a WKNavigationDelegate that overrides DecidePolicy crashes Xamarin.

### Steps to Reproduce

Create a new project with this UIVIewController
```
    public class TestController : UIViewController
    {
        class Navigator : WKNavigationDelegate
        {
            public override void DecidePolicy(WKWebView webView, WKNavigationAction navigationAction, Action<WKNavigationActionPolicy> decisionHandler)
            {
		decisionHandler(WKNavigationActionPolicy.Cancel);
	    }
        }

        WKWebView page;

        void AddPage()
        {
            page?.RemoveFromSuperview();
            page = new WKWebView(new CGRect(0, 0, 400, 400), new WKWebViewConfiguration());
            page.NavigationDelegate = new Navigator();
            page.LoadHtmlString("", new NSUrl(NSBundle.MainBundle.ResourceUrl + ""));

            View.AddSubview(page);

            Console.WriteLine("OK");
        }

        public override void ViewDidLoad()
        {
            UIButton button = new UIButton(new CGRect(200, 500, 0, 0));

            button.SetTitle("Add page", UIControlState.Normal);
            button.SizeToFit();
            button.TouchUpInside += (sender, e) => AddPage();

            View.AddSubview(button);
        }
    }
```
Pressing the button should add an empty webview to the view controller and print "OK" to stdout. This works the first time, but waiting a second and pressing the button again will add the page, print "OK" and then either simply hang or crash the application Sometimes you might get away with more presses, but ultimately it does crash.

### Expected Behavior

App shouldn't crash

### Actual Behavior

App crash

### PR DESCRIPTION
**Pull request by rolfbjarne**
Fixes https://github.com/xamarin/xamarin-macios/issues/4130.

### CODE REVIEW

**Comment by rolfbjarne** [comment]
`do-not-merge` since this PR is only to create an `xcode10` package with the fix for #4130.

File: src/ObjCRuntime/Runtime.cs [hunk 1]
```diff
@@ -1594,6 +1594,21 @@ internal static MethodInfo FindClosedMethod (Type closed_type, MethodBase open_m
 
 			throw ErrorHelper.CreateError (8003, "Failed to find the closed generic method '{0}' on the type '{1}'.", open_method.Name, closed_type.FullName);
 		}
+
+		[EditorBrowsable (EditorBrowsableState.Never)]
```
Comment (Therzok) [inline src/ObjCRuntime/Runtime.cs]: Shouldn't this be in the `#if` block?

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-4867/1/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4867/1/display/redirect) (on internal Jenkins) failed in stage(s) 'Running XM tests on '10.7'' 🔥 : hudson.AbortException: script returned exit code 1

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4867/1//console)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4867/a687249101f68689c1715732194dac7798b26a78/1/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4867/a687249101f68689c1715732194dac7798b26a78/1/jenkins-results/apicomparison/index.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4867/a687249101f68689c1715732194dac7798b26a78/1/jenkins-results/generator-diff/index.html) (please review changes)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.
🔥 [script returned exit code 1](https://jenkins.internalx.com/job/macios/job/PR-4867/1/display/redirect) 🔥

File: src/ObjCRuntime/Runtime.cs (hunk 1 above)
Comment (rolfbjarne) [reply src/ObjCRuntime/Runtime.cs]: No, it applies to either the P/Invoke or the manual implementation.

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4867/2/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4867/2//console)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4867/a687249101f68689c1715732194dac7798b26a78/2/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4867/a687249101f68689c1715732194dac7798b26a78/2/jenkins-results/apicomparison/index.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4867/a687249101f68689c1715732194dac7798b26a78/2/jenkins-results/generator-diff/index.html) (please review changes)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Pull request closed without merging**